spec/02-integration/08-status_api/03-readiness_endpoint_spec.lua

```diff
@@ -0,0 +1,109 @@
+local helpers = require "spec.helpers"
+
+
+for _, strategy in helpers.all_strategies() do
+describe("Status API - with strategy #" .. strategy, function()
+  local client
+  local admin_client
+
+  lazy_setup(function()
+    helpers.get_db_utils(nil, {}) -- runs migrations
+    assert(helpers.start_kong {
+      status_listen = "127.0.0.1:9500",
+      plugins = "admin-api-method",
+      database = strategy,
+      nginx_worker_processes = 8,
+    })
+    client = helpers.http_client("127.0.0.1", 9500, 20000)
+
+    admin_client = helpers.admin_client()
+  end)
+
+  lazy_teardown(function()
+    if client then
+      client:close()
+    end
+    helpers.stop_kong()
+  end)
+ 
+  describe("status readiness endpoint", function()
+    it("should return 503 when no config, return 200 in db mode", function()
+      local res = assert(client:send {
+        method = "GET",
+        path = "/status/ready",
+      })
+
+      assert.res_status(strategy == "off" and 503 or 200, res)
```

Comment:
These if else tests are very confusing.
Can we have a dedicated test for off and another one for traditional/hybrid?
I don't mind the repetition. The testing logic should be straightforward to understand and these conditions make it very easy to write an incorrect test.

What do you think?

Reply:
Ok I'll use seperated tests.